### PR TITLE
fix: validate git URLs

### PR DESCRIPTION
**Pull request by leafty**
See: #712.

What is done with this change:
* Validate the repository URL (when the project is created or updated): the repository URL must be an HTTP or HTTPS URL. SSH is not supported by RenkuLab at the moment.
* Remove exact duplicates.

What is not included with this change:
* Resolve the user-provided URL to a canonical URL (e.g. https://github.com/SwissDataScienceCenter/renku-data-services -> https://github.com/SwissDataScienceCenter/renku-data-services.git). This would require connecting validation code with connected services and is left for a future change.

/deploy #notest renku=release-0.66.0

### CODE REVIEW

File: components/renku_data_services/project/core.py [hunk 1]
```diff
@@ -182,6 +186,29 @@ def validate_session_secrets_patch(
...
+            without_duplicates.append(repo)
+            seen.add(repo)
+    return without_duplicates
+
+
+def _validate_repository(repository: str) -> str:
+    """Validate a git repository."""
+    stripped = repository.strip()
+    parsed = urlparse(stripped)
+    if parsed.scheme != "http" and parsed.scheme != "https":
```
Comment (Panaetius) [inline components/renku_data_services/project/core.py:207]: suggestion(feel free to ignore):
```suggestion
    if parsed.scheme not in ["http", "https"]:
```

File: test/bases/renku_data_services/data_api/test_projects.py [hunk 2]
```diff
@@ -225,6 +225,42 @@ async def test_project_creation_with_conflicting_slug(sanic_client, user_headers
...
 
+@pytest.mark.asyncio
+async def test_project_creation_with_duplicate_repositories(sanic_client, user_headers, regular_user) -> None:
+    namespace = regular_user.namespace.slug
+    payload = {
+        "name": "My Project",
+        "namespace": namespace,
+        "repositories": [
+            "https://github.com/SwissDataScienceCenter/renku-data-services.git",
+            "https://github.com/SwissDataScienceCenter/renku-data-services.git",
```
Comment (Panaetius) [inline test/bases/renku_data_services/data_api/test_projects.py:236]: question: this would probably fail if it was
```suggestion
            "http://github.com/SwissDataScienceCenter/renku-data-services.git",
            "https://github.com/SwissDataScienceCenter/renku-data-services.git",
```
but I'm not sure if we want to handle this case. In a way it's not a duplicate, but I also can't think of a reason why a user would want both http and https for the same repo, so preventing an edge case for them could be nice

Comment (leafty) [reply test/bases/renku_data_services/data_api/test_projects.py:236]: I think it's something that can be handled once we try to re-write repo URLs to use the canonical URL.

**Review by Panaetius — APPROVED** [review]
